lib/src/controls/better_player_controls_state.dart

```diff
@@ -248,26 +250,45 @@ abstract class BetterPlayerControlsState<T extends StatefulWidget>
   }
 
   ///Build both track and resolution selection
-  ///Track selection is used for HLS videos
+  ///Track selection is used for HLS / DASH videos
   ///Resolution selection is used for normal videos
   void _showQualitiesSelectionWidget() {
-    final List<String> trackNames =
+    // HLS
+    final List<String> hlsTrackNames =
         betterPlayerController!.betterPlayerDataSource!.hlsTrackNames ?? [];
-    final List<BetterPlayerHlsTrack> tracks =
-        betterPlayerController!.betterPlayerTracks;
+    final List<BetterPlayerHlsTrack> hlsTracks =
+        betterPlayerController!.betterPlayerHlsTracks;
     final List<Widget> children = [];
-    for (var index = 0; index < tracks.length; index++) {
-      final track = tracks[index];
+    for (var index = 0; index < hlsTracks.length; index++) {
+      final track = hlsTracks[index];
 
       String? preferredName;
       if (track.height == 0 && track.width == 0 && track.bitrate == 0) {
         preferredName = betterPlayerController!.translations.qualityAuto;
       } else {
-        preferredName = trackNames.length > index ? trackNames[index] : null;
+        preferredName = hlsTrackNames.length > index ? hlsTrackNames[index] : null;
       }
-      children.add(_buildTrackRow(tracks[index], preferredName));
+      children.add(_buildTrackRow(hlsTracks[index], preferredName));
     }
 
+    //DASH
+    final List<String> dashTrackNames =
```

Comment:
Seems like you're duplicating same code for dash, Can you create one method which will setup qualities based on passed list of tracks no matter if it's hls or dash?

Reply:
Solved with the unification